odxtools/parameters/systemparameter.py

```diff
@@ -38,18 +44,51 @@ def parameter_type(self) -> ParameterType:
     @property
     @override
     def is_required(self) -> bool:
-        raise NotImplementedError("SystemParameter.is_required is not implemented yet.")
+        # if a SYSTEM parameter is not specified explicitly, its value
+        # can be determined from the operating system if it is type is
+        # predefined
+        return self.sysparam not in PREDEFINED_SYSPARAM_VALUES
 
     @property
     @override
     def is_settable(self) -> bool:
-        raise NotImplementedError("SystemParameter.is_settable is not implemented yet.")
+        return True
 
     @override
     def _encode_positioned_into_pdu(self, physical_value: Optional[ParameterValue],
                                     encode_state: EncodeState) -> None:
-        raise NotImplementedError("Encoding a SystemParameter is not implemented yet.")
+        if physical_value is None:
+            # determine the value to be encoded automatically
+            now = datetime.now()
+            if self.sysparam == "TIMESTAMP":
```

Comment:
Maybe provide users a way to provide a callback for their custom system param, and a way to override default implementation ?

Basically a dict[str, Callback] that users could add to, or change.

Reply:
if you have a custom sysparam, just specify the value of the parameter explicitly. Adding a mechanism for custom callbacks would add much more complexity for no benefit AFAICS (if the parameter is not predefined, `.is_required` will return true...)